sw/device/lib/base/log.c

```diff
@@ -0,0 +1,74 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/log.h"
+
+#include "sw/device/lib/base/print.h"
+
+/**
+ * Converts a severity to a static string.
+ */
+const char *stringify_severity(log_severity_t severity) {
+  switch (severity) {
+    case kLogSeverityInfo:
+      return "INFO";
+    case kLogSeverityWarn:
+      return "WARN";
+    case kLogSeverityError:
+      return "ERROR";
+    default:
+      return "UNKNOWN";
+  }
+}
+
+/**
+ * Extracts the base name from a long filename.
+ */
+static const char *basename(const char *file_name) {
```

Comment:
Just a thought - can we have this handled by meson instead (have meson compute the basename of a C source and add `-D__FILENAME__=<basename(file)>` at compile time)?

Reply:
We could, but it'd be kind of painful, since we'd have to replicate the basename logic in every target... pretty error-prone if you ask me.

It's not a cost DV has to shoulder, anyway, so I wouldn't worry about it.